astparser cannot be built on Solaris due to the lack of strndup()

```
src/ast_scanner.lpp: In function 'char* make_cast(const char*)':
src/ast_scanner.lpp:172:36: error: 'strndup' was not declared in this scope
make[2]: *** [modules/astparser/CMakeFiles/astparser.dir/ast_scanner.cpp.o] Error 1
```

--- a/modules/astparser/src/ast_scanner.lpp
+++ b/modules/astparser/src/ast_scanner.lpp
@@ -169,8 +169,14 @@ static bool isRegexSubstModifier(ASTRegexSubstExpression* rs, int c) {
 
 static char* make_cast(const char* s) {
     size_t size = strlen(s+5);
-    char* str = strndup(s+5, --size); // copy and cut off the trailing '>'
-    while (size > 0 && str[--size] == ' ') str[size] = '\0'; // trim
+
+    // copy and cut off the trailing '>'
+    char* str = (char*) malloc(size--);
+    memcpy(str, s+5, size);
+    str[size] = '\0';
+
+    // trim
+    while (size > 0 && str[--size] == ' ') str[size] = '\0';
     return str;
 }
 

